External/Plugins/ProjectManager/Helpers/LineEntryDialog.cs

```diff
@@ -172,18 +171,22 @@ private void btnCancel_Click(object sender, System.EventArgs e)
             this.Close();
         }
 
-        void OnLineBoxOnKeyDown(object sender, KeyEventArgs args)
+        void LineBox_KeyDown(object sender, KeyEventArgs e)
         {
-            string selectedText = lineBox.SelectedText;
-            if (string.IsNullOrEmpty(selectedText)) return;
-            Keys keys = args.KeyData;
-            if (keys == shortcutToLowercase) selectedText = selectedText.ToLower();
-            else if (keys == shortcutToUppercase) selectedText = selectedText.ToUpper();
-            else return;
-            int selectionStart = lineBox.SelectionStart;
-            int selectionLength = lineBox.SelectionLength;
-            lineBox.Paste(selectedText);
-            SelectRange(selectionStart, selectionLength);
+            string shortcutId;
+            if (!shortcuts.TryGetValue(e.KeyData, out shortcutId)) return;
+            switch (shortcutId)
+            {
+                case "EditMenu.ToLowercase":
+                case "EditMenu.ToUppercase":
+                    string text = lineBox.SelectedText;
+                    if (string.IsNullOrEmpty(text)) break;
```

Comment:
I made it that way because more shortcut options may be available which might work even when no text is present, an example might be Undo/Redo (I haven't done any other implementations however, since I was meant to only make changes to the structure). If the user customized these shortcuts in the main keyboard settings, they might as well want them to work on these controls.